core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -338,15 +347,10 @@ oppia.factory('oppiaPlayerService', [
             StatsReportingService.recordStateTransition(
               oldStateName, newStateName, answer,
               LearnerParamsService.getAllParams());
-
-            if (exploration.isStateTerminal(newStateName)) {
-              StatsReportingService.recordExplorationCompleted(
-                newStateName, LearnerParamsService.getAllParams());
-            }
           }
 
           $rootScope.$broadcast('updateActiveStateIfInEditor', newStateName);
-          $rootScope.$broadcast('playerStateChange');
+          $rootScope.$broadcast('playerStateChange', newStateName);
```

Comment:
I can't say I'm a fan of broadcasting a new event. Why not just have both cases call an internal function? They're all in the same service, after all.